server/resources/GmailResources.py

```diff
@@ -0,0 +1,159 @@
+
+    This function stores the provided OAuth 2.0 credentials using the user ID as
+    key.
+
+    Args:
+      user_id: User's ID.
+      credentials: OAuth 2.0 credentials to store.
+      user_email: User's email address
+    """
+    new_credentials = GmailCredentials(
+        user_email=user_email,
+        user_gmail_id=user_id,
+        user_gmail_credentials=credentials.to_json()
+    )
+    new_credentials.save_to_db()
+
+
+def exchange_code(authorization_code):
+    """Exchange an authorization code for OAuth 2.0 credentials.
+
+    Args:
+      authorization_code: Authorization code to exchange for OAuth 2.0
+                          credentials.
+    Returns:
+      oauth2client.client.OAuth2Credentials instance.
+    Raises:
+      CodeExchangeException: an error occurred.
+    """
+    flow = flow_from_clientsecrets(CLIENTSECRETS_LOCATION, ' '.join(SCOPES))
+    flow.redirect_uri = REDIRECT_URI
+    try:
+        credentials = flow.step2_exchange(authorization_code)
+        return credentials
+    except FlowExchangeError as e:
+        logging.error('An error occurred: %s', e)
+        raise CodeExchangeException(None)
+
+
+def get_user_info(credentials):
+    """Send a request to the UserInfo API to retrieve the user's information.
+
+    Args:
+      credentials: oauth2client.client.OAuth2Credentials instance to authorize the
+                   request.
+    Returns:
+      User information as a dict.
+    """
+    user_info_service = build(
+        serviceName='oauth2', version='v2',
+        http=credentials.authorize(httplib2.Http()))
+    user_info = None
+    try:
+        user_info = user_info_service.userinfo().get().execute()
+    except httplib2.HttpLib2Error as e:
+        logging.error('An error occurred: %s', e)
+    if user_info and user_info.get('id'):
+        return user_info
+    else:
+        raise NoUserIdException()
+
+
+def get_authorization_url(state):
+    """Retrieve the authorization URL.
+
+    Args:
+      state: State for the authorization URL.
+    Returns:
+      Authorization URL to redirect the user to.
+    """
+    flow = flow_from_clientsecrets(CLIENTSECRETS_LOCATION, ' '.join(SCOPES))
+    flow.params['access_type'] = 'offline'
+    flow.params['approval_prompt'] = 'force'
+    flow.params['state'] = state
+    return flow.step1_get_authorize_url(REDIRECT_URI)
+
+
+class GetAuthURL(Resource):
+    def get(self):
+        # will handle state later, need to get this working for now
+        state=None
+        auth_url = get_authorization_url(state)
+        return {'auth_url': auth_url}, 200
+
+
+class Authorize(Resource):
+    @jwt_required
+    def get(self):
+        data = authorize_parser.parse_args()
+        credentials = exchange_code(data['code'])
+        user_info = get_user_info(credentials)
+        user_id = user_info.get('id')
+        user_email = get_jwt_identity()
+        store_credentials(user_id, credentials, user_email)
+        return {'success': True}, 200
```

Comment:
Because we were asked to change jwt identity from user email to user id, 
get_jwt_identity()  now returns the user id, 
since you already have user_info, maybe just user_info.get('email') should return email?